### PR TITLE
gonic: 0.11.0 -> 0.12.0

### PR DESCRIPTION
**Pull request by lxea**
###### Motivation for this change
0.12.0 released

Also removed the ffmpeg patch as now it just looks on the path for ffmpeg

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Can we use buildGoModule for this?

**Comment by lxea** [comment]
> Can we use buildGoModule for this?

Yeah I'm happy to change it to use it. I actually had a version with it too, I wasn't sure if i should make the change though based on this thread https://github.com/NixOS/nixpkgs/issues/84826 so left it as is.

File: pkgs/servers/gonic/default.nix [hunk 1]
```diff
@@ -7,53 +7,31 @@
...
-  # update these, then run `updateScript` to update dependencies
-  version = "0.11.0";
-  rev = "056fb54a703ef5b5194ce112cbbdd8fb53dbb1ea";
-  sha256 = "0hd794wrz29nh89lfnq67w1rc23sg085rqf1agwlgpqycns2djl9";
-
+  version = "0.12.0";
+  pname = "gonic-${version}";
+in
+buildGoModule {
+  inherit pname version;
```
Comment (SuperSandro2000) [inline pkgs/servers/gonic/default.nix:19]:
```suggestion
buildGoModule rec {
  pname = "gonic";
  version = "0.12.0";
```

File: pkgs/servers/gonic/default.nix [hunk 2]
```diff
@@ -7,53 +7,31 @@
...
 # in the configuration if you disable transcodingSupport.
-, transcodingSupport ? true, ffmpeg
-
-# udpater
-, writers, vgo2nix }:
+, transcodingSupport ? true, ffmpeg }:
 
 assert transcodingSupport -> ffmpeg != null;
 
 let
```
Comment (SuperSandro2000) [inline pkgs/servers/gonic/default.nix:14]:
```suggestion
```

**Comment by SuperSandro2000** [comment]
> I'm happy to make the change to use `buildGoModule` and I had a version that did use it but i wasnt sure what the preference was after reading #84826 so I left it as is.

For now we just use buildGoModule because it is easier to use.

**Pull request merged**